llama_index/vector_stores/postgres.py

```diff
@@ -216,11 +268,132 @@ async def _aquery_with_score(
                         node_id=item.node_id,
                         text=item.text,
                         metadata=item.metadata_,
-                        similarity=(1 - distance),
+                        similarity=(1 - distance) if distance is not None else 0,
                     )
                     for item, distance in res.all()
                 ]
 
+    def _rerank(self, query, results):
```

Comment:
Thoughts on whether this should be handled separately by a Node post processor like [one of our ReRanker modules](https://gpt-index.readthedocs.io/en/stable/examples/node_postprocessor/CohereRerank.html)?

My performance concern with this implementation is that it has the potentially expensive call to `encoder.predict`. For async execution flows, that may be a problem. May make sense to do re-ranking as a customizable post-processing step where they can choose the type of re-ranking to do. Granted, I think we probably need to improve async support for node post processing 😅

Reply:
Interesting, @Disiok had similar feedback when discussing my approach. The reason we ultimately settled on performing the reranking step inside the postgres module at the time was to be consistent with the interface presented by `pinecone` which has native hybrid query support and therefore returns already-ranked results. I wasn't aware of the performance issues related to crossrank functionality (assuming cpu bound?). I can look more into the option you suggested (the only concern is that a user would need to remember the usage distinction between pinecone vs postgres-backed vector stores)

Reply:
Gotcha, thanks for writing out your reasoning. I personally like the de-coupling we'd get with the approach of putting re-ranking into a node post processor. It's possible that folks will just want to use hybrid search with a small `k` lookup value and no re-ranking. Or they may want to make use of Cohere's re-ranker for example. So I'd lean towards the de-coupled approach to maximize flexibility. @Disiok thoughts here?